redirect/redirect.go

```diff
@@ -8,11 +8,24 @@ import (
 )
 
 // SecureRedirectMiddleware redirects the client to the identical URL served via HTTPS
-type SecureRedirectMiddleware struct{}
+type SecureRedirectMiddleware struct {
+	WhiteListedPaths map[string]struct{}
```

Comment:
By only suggestion is on naming. I know this field functions as a whitelist but it seems like the name could better describe what purpose the white list serves. So something like
```
  SkippedPaths map[string]struct{}
```
or 
```
  AllowedInsecurePaths map[string]struct{}
```